posthog/session_recordings/replay_active_users/replay_active_users_query_runner.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime, timedelta
+from typing import Optional
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select
+from posthog.hogql.query import execute_hogql_query
+from posthog.hogql_queries.query_runner import AnalyticsQueryRunner
+from posthog.schema import (
+    CachedReplayActiveUsersQueryResponse,
+    ReplayActiveUsersQuery,
+    ReplayActiveUsersQueryResponse,
+)
+
+
+class ReplayActiveUsersQueryRunner(AnalyticsQueryRunner[ReplayActiveUsersQueryResponse]):
+    query: ReplayActiveUsersQuery
+    cached_response: CachedReplayActiveUsersQueryResponse
+
+    def cache_target_age(self, last_refresh: Optional[datetime], lazy: bool = False) -> Optional[datetime]:
+        if last_refresh is None:
+            return None
+
+        return last_refresh + timedelta(hours=1)
+
+    def to_query(self) -> ast.SelectQuery | ast.SelectSetQuery:
+        # Use Python's datetime.now() which respects frozen time in tests
+        now = datetime.now()
+
+        query = """
+            WITH
+            counted_sessions AS (
+                SELECT
+                    session_id,
+                    any(distinct_id) AS sess_di,
+                    count() AS c
+                FROM raw_session_replay_events
+                WHERE min_first_timestamp >= {python_now} - interval 7 day
+                  AND min_first_timestamp <= {python_now}
+                GROUP BY session_id
+                HAVING date_diff('second', min(min_first_timestamp), max(max_last_timestamp)) > 5
+            ),
+
+            session_persons AS (
+                SELECT
+                    $session_id as session_id,
+                    any(person_id) as person_id,
+                    any(person.properties) as pp
+                FROM events
+                WHERE timestamp >= {python_now} - interval 7 day
+                  AND timestamp <= {python_now}
+                  AND $session_id IN (SELECT session_id FROM counted_sessions)
+                  AND event IN ('$pageview', '$screen', '$autocapture', '$feature_flag_called', '$pageleave', '$identify', '$web_vitals', '$set', 'Application Opened', 'Application Backgrounded')
```

Comment:
these are the most likely events to be on events in the 3 hours before I opened the PR
it should cover most accounts
but is way faster than querying the events table without specifying any events